From 50d9786fb3c6c7fb878886284da3f34d1969061a Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Wed, 6 Sep 2017 22:25:22 +0800
Subject: [PATCH 3315/5242] MLK-17491-15 dts: imx7ulp: add necessary clock for
 gpio node

commit  9dc495b3f702ba1ccd29273c13eefa136f8bc3e6 from
https://source.codeaurora.org/external/imx/linux-imx.git

On MX7ULP, GPIO controller needs two necessary clocks:
Port module clock and GPIO module clock.

Reviewed-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp.dtsi |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index 38f57c9..3282204 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -530,6 +530,9 @@
 			interrupts = <GIC_SPI 48 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-controller;
 			#interrupt-cells = <2>;
+			clocks = <&clks IMX7ULP_CLK_RGPIO2P1>,
+				 <&clks IMX7ULP_CLK_PCTLC>;
+			clock-names = "port", "gpio";
 			gpio-ranges = <&iomuxc1 0 0 32>;
 		};
 
@@ -541,6 +544,9 @@
 			interrupts = <GIC_SPI 49 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-controller;
 			#interrupt-cells = <2>;
+			clocks = <&clks IMX7ULP_CLK_RGPIO2P1>,
+				 <&clks IMX7ULP_CLK_PCTLD>;
+			clock-names = "port", "gpio";
 			gpio-ranges = <&iomuxc1 0 32 32>;
 		};
 
@@ -552,6 +558,9 @@
 			interrupts = <GIC_SPI 50 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-controller;
 			#interrupt-cells = <2>;
+			clocks = <&clks IMX7ULP_CLK_RGPIO2P1>,
+				 <&clks IMX7ULP_CLK_PCTLE>;
+			clock-names = "port", "gpio";
 			gpio-ranges = <&iomuxc1 0 64 32>;
 		};
 
@@ -563,6 +572,9 @@
 			interrupts = <GIC_SPI 51 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-controller;
 			#interrupt-cells = <2>;
+			clocks = <&clks IMX7ULP_CLK_RGPIO2P1>,
+				 <&clks IMX7ULP_CLK_PCTLF>;
+			clock-names = "port", "gpio";
 			gpio-ranges = <&iomuxc1 0 96 32>;
 		};
 
-- 
1.7.9.5

